### ISSUE TITLE
Stack Heap Collision

### ISSUE THREAD

**Issue by Kyrellyan** [OpenCollarTeam/OpenCollar#710]
What version of OpenCollar are you using?:  8.1.0000

What behavior did you expect?: Normal function.  Was sitting and chatting with friends and suddenly got a notification.

What behavior did you see instead?:
[13:08] **RE** Simone Heart Collar [script:oc_core] Script run-time error
[13:08] Stack-Heap Collision
[13:09] **RE** Simone Heart Collar v1.0: oc_core has been reset. If the script stack heaped, please file a bug report on our github.
[13:09] **RE** Simone Heart Collar v1.0: Startup in progress... be patient
[13:09] **RE** Simone Heart Collar v1.0: Collar ready. Startup complete

What steps does someone need to take to reproduce the problem?

**Comment by SilkieSabra** [comment]
Unusual but do update to the current version 8.2.1 and see if it happens again

**Comment by Nast21** [comment]
Hi, I have exactly the same problem but its a new collar on 8.2.1000. I have 2 owners and 21 trusted. Is that the problem?

**Comment by Kyrellyan** [comment]
> Hi, I have exactly the same problem but its a new collar on 8.2.1000. I have 2 owners and 21 trusted. Is that the problem?

Doubtful, I have one owner and no trusted on mine.

**Comment by Medea-Destiny** [comment]
We're running close to the limit on script memory in a bunch of scripts and occasionally there's going to be a stack/heap collision in weird circumstances. If this is a one-off I wouldn't be too concerned. 

Moving Weld to an app rather than having it built-in has been suggested and this would relieve a bit of the load on oc_core.

**Comment by Medea-Destiny** [comment]
@Nast21 Was it oc_core or oc_extension that stack-heaped in your case? 21 trusted is a LOT and will certainly make stack-heap collisions more likely.

@Kyrellyan Was this a one-off, or have you seen it happening since?

**Comment by Nast21** [comment]
Hi, Here is the message I get - every time I log in:
[15:59] OpenCollar Stainless [script:oc_rlvextension] Script run-time error
[15:59] Stack-Heap Collision

**Comment by Kyrellyan** [comment]
@Medea-Destiny So far it seems to have been a one-off.  I did update to 8.2.1 as @SilkieSabra suggested, so not sure if that fixed it or what.  Wish I knew what had caused it to begin with.

**Comment by Medea-Destiny** [comment]
@Nast21 Thanks, yes this is oc_extension not oc_core as I suspected. Personally I only tested the current version with a dozen people on the trusted/owner lists, you're running with twice as many and I'm sure that's the issue. The script is very tight for memory. Startup hammers the scripts so this is the time when it's most likely to get a stack-heap collision. 

@nirea If you recall you asked me to add integer constants for bitmasks values to the oc_rlvextension script, which is an additional squeeze on memory. While I appreciate the ease of comprehension this adds, can I suggest we try reverting that and using comments to make things clear instead? This is one of those scripts where every byte counts.
